frontend/src/pages/analytics/segments/UserSegmentDetails.tsx

```diff
@@ -0,0 +1,616 @@
+
+  const breadcrumbItems = [
+    {
+      text: t('breadCrumb.name'),
+      href: '/',
+    },
+    {
+      text: t('breadCrumb.segments'),
+      href: `/analytics/${projectId}/app/${appId}/segments`,
+    },
+    {
+      text: segmentId,
+      href: `/analytics/${projectId}/app/${appId}/segments/${segmentId}`,
+    },
+  ];
+
+  useEffect(() => {
+    const getSegmentDetails = async () => {
+      setIsLoading(true);
+
+      const segmentApiResponse = await getSegmentById({
+        appId: appId,
+        segmentId: segmentId,
+      });
+      if (segmentApiResponse.success) {
+        const segment: SegmentDdbItem = segmentApiResponse.data;
+        console.log(segment);
+        setSegment(segment);
+      }
+
+      setIsLoading(false);
+    };
+
+    getSegmentDetails();
+  }, []);
+
+  useEffect(() => {
+    getSegmentJobHistory();
+    getSegmentSampleData();
+  }, []);
+
+  const getSegmentJobHistory = async () => {
+    setIsJobsLoading(true);
+
+    const segmentJobsResponse = await getSegmentJobs({ segmentId });
+    if (segmentJobsResponse.success) {
+      setSegmentJobs(segmentJobsResponse.data as SegmentJobStatusItem[]);
+    }
+
+    setIsJobsLoading(false);
+  };
+
+  const getSegmentSampleData = async () => {
+    setIsSampleDataLoading(true);
+
+    const sampleDataResponse = await getSampleData({ segmentId });
+    if (sampleDataResponse.success) {
+      setSegmentSample(sampleDataResponse.data as SegmentJobStatusItem);
+      console.log(sampleDataResponse.data);
+    }
+
+    setIsSampleDataLoading(false);
+  };
+
+  const constructSegmentTrendChart = () => {
+    const getTimeLabel = (time: number) => new Date(time).toLocaleString();
+    const jobs = segmentJobs.filter((job) => job.jobStatus === 'Completed');
+
+    return (
+      <MixedLineBarChart
+        series={
+          jobs.length === 0
+            ? []
+            : [
+                {
+                  title: defaultStr(
+                    t('analytics:segment.details.segmentUserNumber')
+                  ),
+                  type: 'bar',
+                  data: jobs.map((job) => ({
+                    x: getTimeLabel(job.jobStartTime),
+                    y: job.segmentUserNumber,
+                  })),
+                },
+                {
+                  title: defaultStr(
+                    t('analytics:segment.details.totalUserNumber')
+                  ),
+                  type: 'line',
+                  data: jobs.map((job) => ({
+                    x: getTimeLabel(job.jobStartTime),
+                    y: job.totalUserNumber,
+                  })),
+                },
+              ]
+        }
+        xDomain={jobs.map((job) => getTimeLabel(job.jobStartTime))}
+        xScaleType="categorical"
+        xTitle={defaultStr(t('analytics:segment.details.date'))}
+        yTitle={defaultStr(t('analytics:segment.details.userNumber'))}
+        hideFilter
+        statusType={isJobsLoading ? 'loading' : 'finished'}
```

Comment:
use  `ternary` from utils

Reply:
Change it to use `ternary`.

@chenhaiyun  I was wondering should we always use `ternary()` utils instead of ternary operator even the condition and expressions are pretty straightforward?